src/app/triggerFromForm.ts

```diff
@@ -0,0 +1,112 @@
+import {Tickets} from 'node-zendesk';
+
+import {AppContext, AppCallValues} from 'mattermost-redux/types/apps';
+
+import {SubscriptionFields, TriggerFields} from '../utils/constants';
+
+interface ITriggerFromFrom {
+    getTrigger(): Tickets.CreatePayload;
+}
+
+export class TriggerFromForm implements ITriggerFromFrom {
+    values: AppCallValues;
+    context: AppContext;
+    trigger: any
+
+    constructor(context: AppContext, values: AppCallValues) {
+        this.values = values;
+        this.context = context;
+        this.trigger = {};
+    }
+
+    getTrigger(): any {
+        // If not a new trigger, add the trigger ID to the payload
+        // This is a signal to update the trigger, not create a new one
+        if (!this.isNewTrigger()) {
+            const subID = this.values[SubscriptionFields.SubSelectName].value;
+            this.addField('id', subID);
+        }
+        this.addTitle();
+        this.addDescription();
+        this.addActions();
+        this.addConditions();
+
+        return {trigger: this.trigger};
+    }
+
+    addActions(): void {
+        const actions = [
+            {
+                field: TriggerFields.ActionField,
+                value: [
+                    TriggerFields.TargetID,
+                    this.getJSONDataFields(),
+                ],
+            },
+        ];
+        this.addField('actions', actions);
+    }
+
+    // getJSONDataFields constructs the object text string for a trigger
+    getJSONDataFields(): string {
+        const channelID = this.values[SubscriptionFields.ChannelPickerSelectName].value;
+        const pairs = TriggerFields.ActionValuePairs;
+        pairs[TriggerFields.ChannelIDKey] = channelID;
+        return JSON.stringify(pairs);
+    }
+
+    getChannelIDobject(): {} {
+        const fieldName = SubscriptionFields.ChannelPickerSelectName;
+        const channelID = this.values[fieldName].value;
+        return {fieldName: channelID};
+    }
+
+    addTitle(): void {
+        let title = SubscriptionFields.PrefixTriggersTitle;
+        title += this.context.channel_id;
+        title += ' ' + this.values[SubscriptionFields.SubTextName];
+        this.addField('title', title);
+    }
+
+    addDescription(): void {
+        const description = String(this.values[SubscriptionFields.SubTextName]);
+        this.addField('description', description);
+    }
+
+    addField(key: string, value: any): void {
+        this.trigger[key] = value;
+    }
+
+    isNewTrigger(): boolean {
+        const subID = this.values[SubscriptionFields.SubSelectName].value;
+        return subID === SubscriptionFields.NewSub_OptionValue;
+    }
+
+    addConditions(): void {
+        const conditions = {
+            any: [],
```

Comment:
Can we set the type here?

Reply:
put into catch-all NITs ticket https://mattermost.atlassian.net/browse/MM-33787